### PR TITLE
dependencies api: Add Syncer interface

### PR DESCRIPTION
**Pull request by efritz**
This removes the import of `cmd/frontend/backend` from `internal/codeintel`.

## Test plan

Behavior-preserving refactor.

### CODE REVIEW

File: internal/search/repos/repos.go [hunk 1]
```diff
@@ -582,6 +583,15 @@ func (r *Resolver) dependencies(ctx context.Context, op *search.RepoOptions) (_
 	return depNames, depRevs, nil
 }
 
+type syncer struct {
+	repoStore database.RepoStore
+}
+
+func (s *syncer) Sync(ctx context.Context, repo api.RepoName) error {
+	_, err := backend.NewRepos(s.repoStore).GetByName(ctx, repo)
```
Comment (tsenart) [inline internal/search/repos/repos.go:591]: nit: We could avoid calling backend.NewRepos on every Sync call by putting the return type as a field in syncer.

File: internal/codeintel/dependencies.go [hunk 2]
```diff
@@ -31,38 +29,42 @@ import (
 // and package levels.
 type DependenciesService struct {
 	db              database.DB
-	sync            func(context.Context, api.RepoName) (*types.Repo, error)
+	syncer          Syncer
 	lockfileService *lockfiles.Service
 	operations      *dependencyServiceOperations
 }
 
+type Syncer interface {
```
Comment (tsenart) [inline internal/codeintel/dependencies.go:37]: Small comment would help here.

**Review by tsenart — APPROVED** [review]

File: internal/codeintel/dependencies.go [hunk 3]
```diff
@@ -31,38 +29,42 @@ import (
...
+	Sync(context.Context, api.RepoName) error
+}
+
 var (
 	depSvc     *DependenciesService
 	depSvcOnce sync.Once
 )
 
-func GetOrCreateGlobalDependencyService(db database.DB, repoStore database.RepoStore) *DependenciesService {
+func GetOrCreateGlobalDependencyService(db database.DB, syncer Syncer) *DependenciesService {
```
Comment (tsenart) [inline internal/codeintel/dependencies.go:46]: E_NAME_TOO_LONG

I'll stop, I promise.

File: internal/codeintel/dependencies.go (hunk 2 above)
Comment (efritz) [reply internal/codeintel/dependencies.go:37]: Updated.

File: internal/search/repos/repos.go (hunk 1 above)
Comment (efritz) [reply internal/search/repos/repos.go]: Updated.

**Pull request merged**